http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/NettyRequestExecutor.java

```diff
@@ -87,8 +89,8 @@ public NettyRequestExecutor(RequestContext context) {
     @SuppressWarnings("unchecked")
     public CompletableFuture<Void> execute() {
         Promise<Channel> channelFuture = context.eventLoopGroup().next().newPromise();
+        executeFuture = createExecutionFuture(channelFuture);
```

Comment:
The switch was to make sure the get-metrics call was before the connection acquire call, so that we'd always get metrics if the connection acquire failed. Without the switch, if the connection acquire fails we'd sometimes not get metrics.

Now, there's one additional task in the event loop before connection acquisition and we are creating the future before adding the acquire task to the event loop. That is technically going to increase latency, but I don't expect it to be any amount that we'd notice.

Reply:
We can go with that, I'm not too afraid of latency.